auth0/resource_auth0_client.go

```diff
@@ -465,6 +466,12 @@ func createClient(d *schema.ResourceData, m interface{}) error {
 func readClient(d *schema.ResourceData, m interface{}) error {
 	api := m.(*management.Management)
 	c, err := api.Client.Read(d.Id())
+
+	if err != nil && strings.HasPrefix(err.Error(), "404") {
+		d.SetId("")
+		return nil
+	}
+
 	if err != nil {
 		return err
 	}
```

Comment:
Checking for the error string is not a safe way to check this type of error as its not guaranteed to always be present.

We should instead type assert whether `err` is a [`management.Error`](https://github.com/go-auth0/auth0/blob/master/management/management.go#L274-L277) - which it should be - and check the status from that.

```
e, err := api. Client.Read(d.Id())
if err != nil {
	if mErr, ok := err.(management.Error); ok {
		if mErr.Status() == http.StatusNotFound {
			d.SetId("")
			return nil
		}
	}
	return err
}
```

Reply:
Yes...I looked at the unexported `managementError` struct, and thought the best we could do was to read the status code from the string.  I missed the fact that the struct implements the exported `Error` interface.  I'll make that fix.